src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,71 +1,48 @@
 public class CartController {
+    private final CartService cartDbService;
+    private final ProductMapper productMapper;
 
-    List<CartDto> CartDtoList = new ArrayList<CartDto>(){
-        {
-            add(new CartDto(1L, "kurtka zimowa", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(100), 1L));
-            add(new CartDto(2L, "płaszcz", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(150), 1L));
-            add(new CartDto(8L,"krawat", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(50), 2L));
-
-        }
-    };
 
-    @GetMapping
-    public ResponseEntity<List<CartDto>> getCart(){
-        return ResponseEntity.ok(CartDtoList);
+    @PostMapping("{userId}")
+    public void createEmptyCart(@PathVariable("userId") Long userId) {
+        cartDbService.createEmptyCart(userId);
     }
 
-    @GetMapping(value = "{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long cartId){
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                return ResponseEntity.ok(cart);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @GetMapping("{userId}")
+    public List<ProductDto> getCart(@PathVariable("userId") Long userId) {
+        return productMapper.mapToProductDtoList(cartDbService.getUserCart(userId));
     }
 
-    @DeleteMapping(value = "{cartId}")
-    public ResponseEntity<Void> deleteCart(@PathVariable Long cartId) {
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                CartDtoList.remove(cart);
-                return ResponseEntity.ok().build();
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @PutMapping("{userId}/{productId}")
+    public void addItemToCart(@PathVariable("userId") Long userId, @PathVariable Long productId) {
+        cartDbService.addItemToCart(userId, productId);
     }
 
-    @PutMapping(path="{cartId}",consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<CartDto> updateCart(@PathVariable Long cartId, @RequestBody CartDto cartDto) {
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)) {
-                CartDtoList.remove(cart);
-                CartDtoList.add(cartDto);
-                return ResponseEntity.ok(cartDto);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @DeleteMapping("{userId}/{productId}")
+    public void deleteItemFromCart(@PathVariable("userId") Long userId, @PathVariable("productId") Long productId){
+        cartDbService.deleteItemFromCart(userId, productId);
     }
 
-    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Void> createCart(@RequestBody CartDto cartDto) {
-        CartDtoList.add(cartDto);
-        return ResponseEntity.ok().build();
+    @PostMapping("/{userId}/createOrder")
```

Comment:
It's not a problem but it seems more natural to me to swap the order:
"/createOrder/{userId}"